Formula/hwloc.rb

```diff
@@ -19,16 +19,26 @@ class Hwloc < Formula
   end
 
   depends_on "pkg-config" => :build
-  depends_on "cairo" => :optional
+  depends_on "cairo" => [:optional, "with-x11"]
```

Comment:
Using dependencies that require options is deprecated. See https://github.com/Homebrew/homebrew-core/issues/13133

Reply:
There is no way to get GUI support from `lstopo` without `cairo` being built with X11 support. `hwloc --with-cairo` only buys you some extra image output options, but if you want to look at an exotic remote system, such as a Xeon Phi KNL in different memory modes, you probably want to explore it in the `lstopo` GUI locally after generating the `.xml` file on the remote machine.

Reply:
Could we add a caveat that cairo is needed with x11 support to be able to use the lstopo GUI? Then test if we have `cairo --with-x11` to change the `--with-x` configure flag?

Reply:
(I presume there's no hope of getting cairo to default to `--with-x11`)

Reply:
@zfs If I change this to just check whether or not cairo was build with X11 support to determine whether to pass `--with-x11` or `--without-x11` (possibly with a caveat section added) would this be an acceptable enahncement?

Also, what is the correct syntax for testing if a package was built with an option? `build.with? "cairo" => ["with-x11"]`?